test/conformance/ingress/httpoption.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"knative.dev/networking/pkg/apis/networking"
+	"knative.dev/networking/pkg/apis/networking/v1alpha1"
+	"knative.dev/networking/test"
+)
+
+// TestHTTPOption verifies that the Ingress properly handles HTTPOption field.
+func TestHTTPOption(t *testing.T) {
+	// net-istio cannot support parallel option as one HTTPOption effects globally.
+	// https://github.com/knative-sandbox/net-istio/issues/637
+	// t.Parallel()
+	ctx, clients := context.Background(), test.Setup(t)
+
+	tests := []struct {
+		name       string
+		httpOption v1alpha1.HTTPOption
+		codes      []int
+	}{{
+		name:       "enabled",
+		httpOption: v1alpha1.HTTPOptionEnabled,
+		codes:      []int{http.StatusOK},
+	}, {
+		name:       "redirected",
+		httpOption: v1alpha1.HTTPOptionRedirected,
+		codes:      []int{http.StatusMovedPermanently /*301*/, http.StatusFound /*302*/},
```

Comment:
Really just out of interest: do we ever/would we ever/should we ever return a 302 here? My (very shaky) understanding was a http->https redirect would/should pretty much always be 301

Reply:
Good catch. Yes, agree. 301 is almost always used for http to https redirect. 
I updated this PR as well as comment in config-network by https://github.com/knative/networking/pull/416.